### PR TITLE
Fix package paths for OSX and Ubuntu.

### LINKED ISSUE #477: Frequent test failures on OSX and Ubuntu
The following tests fail frequently (but not always) on OSX and Ubuntu for Preview1:

```
[22:54:39]	[Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test.MvcControllerTest.TestControllerGenerators(baselineFile: "EmptyController.txt", generatedFilePath: "EmptyController.cs", args: ["aspnet-codegenerator", "-p", ".", "controller", "--controllerName", ...])] Assert.True() Failure
Expected: True
Actual:   False
   at Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test.E2ETestBase.VerifyFileAndContent(String generatedFilePath, String baselineFile) in /datadrive/33bdfc1cae7b2a38/.r/Scaffolding/test/E2E_Test/E2ETestBase.cs:line 54
   at Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test.MvcControllerTest.TestControllerGenerators(String baselineFile, String generatedFilePath, String[] args) in /datadrive/33bdfc1cae7b2a38/.r/Scaffolding/test/E2E_Test/MvcControllerTest.cs:line 48

[22:54:39]	[Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test.AreaScaffolderTests.TestAreaGenerator] Assert.True() Failure
Expected: True
Actual:   False
   at Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test.E2ETestBase.VerifyFileAndContent(String generatedFilePath, String baselineFile) in /datadrive/33bdfc1cae7b2a38/.r/Scaffolding/test/E2E_Test/E2ETestBase.cs:line 54
   at Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test.AreaScaffolderTests.TestAreaGenerator() in /datadrive/33bdfc1cae7b2a38/.r/Scaffolding/test/E2E_Test/AreaScaffolderTests.cs:line 47

[22:54:56]	[Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test.MvcControllerTest.TestControllerGenerators(baselineFile: "EmptyController_withrelativepath.txt", generatedFilePath: "Controllers/EmptyController.cs", args: ["aspnet-codegenerator", "-p", ".", "controller", "--controllerName", ...])] Assert.True() Failure
Expected: True
Actual:   False
   at Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test.E2ETestBase.VerifyFileAndContent(String generatedFilePath, String baselineFile) in /datadrive/33bdfc1cae7b2a38/.r/Scaffolding/test/E2E_Test/E2ETestBase.cs:line 54
   at Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test.MvcControllerTest.TestControllerGenerators(String baselineFile, String generatedFilePath, String[] args) in /datadrive/33bdfc1cae7b2a38/.r/Scaffolding/test/E2E_Test/MvcControllerTest.cs:line 48

[22:55:14]	[Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test.MvcControllerTest.TestControllerGenerators(baselineFile: "ReadWriteController.txt", generatedFilePath: "ActionsController.cs", args: ["aspnet-codegenerator", "-p", ".", "controller", "--controllerName", ...])] Assert.True() Failure
Expected: True
Actual:   False
   at Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test.E2ETestBase.VerifyFileAndContent(String generatedFilePath, String baselineFile) in /datadrive/33bdfc1cae7b2a38/.r/Scaffolding/test/E2E_Test/E2ETestBase.cs:line 54
   at Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test.MvcControllerTest.TestControllerGenerators(String baselineFile, String generatedFilePath, String[] args) in /datadrive/33bdfc1cae7b2a38/.r/Scaffolding/test/E2E_Test/MvcControllerTest.cs:line 48
```

http://aspnetci/viewType.html?buildTypeId=XPlat_Mac_MacUniverseCoreCLR&branch_XPlat=2.0.0-preview1&tab=buildTypeStatusDiv
http://aspnetci/viewType.html?buildTypeId=XPlat_Linux_UbuntuUniverseCoreCLR&branch_XPlat=2.0.0-preview1&tab=buildTypeStatusDiv

### PR DESCRIPTION
**Pull request by prafullbhosale**
Fixes #477 

cc @Eilon @mikeharder @mlorbetske

### CODE REVIEW

**Comment by mikeharder** [comment]
What is the customer impact of this change?  Why do the tests fail sometimes (but not always) without this change?

**Comment by prafullbhosale** [comment]
There should be no customer impact because of this.
Without this change, I have noticed that the package created on OSX has the templates included in a flat directory structure as 
screen shot 2017-05-08 at 10 09 32 am


instead of being like this: 

screen shot 2017-05-08 at 10 50 26 am


On windows the package paths are fine with the existing includes.

**Comment by mikeharder** [comment]
@Eilon or @muratg: What do you think of this change for Preview1?

**Comment by mikeharder** [comment]
@prafullbhosale: Can you clarify how there is no customer impact?  If there's no customer impact to these test failures, why are the tests being run at all?

**Comment by prafullbhosale** [comment]
> Can you clarify how there is no customer impact? 

Let me rephrase, if we build and publish packages on the nuget feeds only from windows, there is no impact as the packages generated on windows with or without this change are the same. 

This looks to me like an inconsistency in `pack` behavior in windows vs OSX. 
The tests that are failing are E2E tests that use newly generated packages from the specific build. On OSX, the packages without this change had an incorrect layout of template files. (see above screenshots).

To be absolutely clear, there *will* be a customer impact without this change. It will be seen if the user builds the repo on mac and uses the packages generated from that build.

**Comment by prafullbhosale** [comment]
Spoke with @Eilon offline on this. 
Will merge this and file another bug to investigate why this started to break recently.
